src/bin/tx_create.rs

```diff
@@ -0,0 +1,127 @@
+extern crate cardano;
+extern crate chain_addr;
+extern crate chain_core;
+extern crate chain_crypto;
+extern crate chain_impl_mockchain;
+extern crate reqwest;
+extern crate serde_json;
+extern crate structopt;
+
+mod utils;
+
+use cardano::util::hex;
+use chain_addr::{Address, Discrimination, Kind};
+use chain_core::property::Serialize;
+use chain_crypto::{Ed25519Extended, PublicKey};
+use chain_impl_mockchain::fee::LinearFee;
+use chain_impl_mockchain::key::SpendingSecretKey;
+use chain_impl_mockchain::transaction::{Input, TransactionId, UtxoPointer};
+use chain_impl_mockchain::txbuilder::{GeneratedTransaction, OutputPolicy, TransactionBuilder};
+use chain_impl_mockchain::value::Value;
+use std::io;
+use structopt::StructOpt;
+use utils::SegmentParser;
+
+fn main() {
+    TxCreate::from_args().exec();
+}
+
+/// Create transaction binary blob
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct TxCreate {
+    /// transaction input. Must have format `<hex-encoded-transaction-id>:<output-index>:<value>`.
+    /// E.g. `1234567890abcdef:2:535`. At least 1 value required.
+    #[structopt(name = "input", short, long, parse(try_from_str = "parse_input"))]
+    inputs: Vec<Input>,
+    /// transaction output. Must have format `<hex-encoded-address>:<value>`.
+    /// E.g. `abcdef1234567890:501`. At least 1 value required.
+    #[structopt(name = "output", short, long, parse(try_from_str = "parse_output"))]
+    outputs: Vec<(Address, Value)>,
```

Comment:
don't use hexadecimal for the address, use bech32

Reply:
Hex in output is to keep consistency with hex in input which is to keep consistency with REST API. We probably should keep APIs consistent. I prefer bech32 as well, so maybe we should change REST API to return bech32s everywhere instead of hexes?

Reply:
yes we need to change that too. We need bech32 here as this is _user_ facing.

Reply:
and input are hex decimal if they are from a utxo pointer. Otherwise they need to be bech32 if it is an account.

Reply:
OTOH the REST API is not supposed to be used by humans anymore, they should get information via CLI and it can convert hexes to bech32 (or push it to REST API itself with `Accept` header)